assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidgetGA4.js

```diff
@@ -0,0 +1,362 @@
+ * @property {string}             serviceURL - Link to relevant Google Analytics page for the report.
+ * @property {boolean}            isLoading  - Loading status for report.
+ * @property {(Object|undefined)} error      - Error object if exists, otherwise undefined.
+ * @return {OverallPageMetricsReport} Analytics report data and state.
+ */
+function useOverallPageMetricsReport() {
+	const viewOnlyDashboard = useViewOnly();
+
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+			compare: true,
+		} )
+	);
+
+	const url = useSelect( ( select ) =>
+		select( CORE_SITE ).getCurrentEntityURL()
+	);
+
+	const args = {
+		...dates,
+		dimensions: [ 'date' ],
+		metrics: [
+			{
+				name: 'screenPageViews',
+			},
+			{
+				name: 'sessions',
+			},
+			{
+				name: 'engagedSessions',
+			},
+			{
+				name: 'averageSessionDuration',
+			},
+		],
+		orderby: [
+			{
+				dimension: {
+					dimensionName: 'date',
+				},
+			},
+		],
+		url,
+	};
+
+	const reportArgs = generateDateRangeArgs( dates );
+
+	if ( isURL( url ) ) {
+		reportArgs[ 'explorer-table.plotKeys' ] = '[]';
+		reportArgs[ '_r.drilldown' ] = `analytics.pagePath:${ getURLPath(
+			url
+		) }`;
+	}
+
+	const isLoading = useSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getReport',
+				[ args ]
+			)
+	);
+
+	const error = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			args,
+		] )
+	);
+
+	const serviceURL = useSelect( ( select ) => {
+		if ( viewOnlyDashboard ) {
+			return null;
+		}
+
+		return select( MODULES_ANALYTICS ).getServiceReportURL(
+			'visitors-overview',
+			reportArgs
+		);
+	} );
+
+	const report = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( args )
+	);
+
+	return {
+		report,
+		dates,
+		serviceURL,
+		isLoading,
+		error,
+	};
+}
+
+/**
+ * Parse Analytics 4 report into data suitable for rendering the data blocks in the Overall Page Metrics widget.
+ *
+ * @typedef {Object} OverallPageMetricsData
+ * @property {string}         metric          - Google Analytics metric identifier.
+ * @property {string}         title           - Translated metric title.
+ * @property {Array.<Object>} sparkLineData   - Data for rendering the sparkline.
+ * @property {string}         [datapointUnit] - Optional datapoint unit, e.g. '%', 's'.
+ * @property {number}         total           - Total count for the metric.
+ * @property {number}         change          - Monthly change for the metric.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} report    Analytics report data.
+ * @param {string} startDate Start date for the report.
+ * @return {Array.<OverallPageMetricsData>} Array of data for rendering the data blocks in the Overall Page Metrics widget.
+ */
+
+function calculateOverallPageMetricsData( report, startDate ) {
```

Comment:
This is a utility function, please, move it to the `utils` folder. I know that the same function exists in the original widget. If you want, feel free to move that function as well.

Reply:
No problem, I've extracted the function is this and the UA version.